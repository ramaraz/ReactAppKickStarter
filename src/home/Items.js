import React from 'react';

const Items = ({items}) => { //destructure items
    return(
        <ul>
          {items.map((item,index) => (
            <li key={Date.now() + 'r' + index}>{item}</li>
          ))}
        </ul>
    )
}
 export default Items;