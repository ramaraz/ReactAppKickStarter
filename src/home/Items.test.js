import React from 'react';
import Items from './Items';
import renderer from 'react-test-renderer';

const props ={
    items: ['item1','item2','item3','item4']
}
//TODO : change this to BDD
describe('<Items />',() =>{
  it('renders items present in th props',() =>{
    const items = renderer
    .create(<Items {...props} />)
    .toJSON();
    expect(items).toMatchSnapshot()
  })
})

