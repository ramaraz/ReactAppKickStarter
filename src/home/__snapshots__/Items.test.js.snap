// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> renders items present in th props 1`] = `
<ul>
  <li>
    item1
  </li>
  <li>
    item2
  </li>
  <li>
    item3
  </li>
  <li>
    item4
  </li>
</ul>
`;
